cpp/src/hash/concurrent_unordered_map.cuh

```diff
@@ -86,6 +86,10 @@ union pair_packer<pair_type, std::enable_if_t<is_packable<pair_type>()>> {
 /**
  * Supports concurrent insert, but not concurrent insert and find.
  *
+ * @note User is responsible for following stream semantics:
+ * - Either same stream should be used to create the map as is used by the kernels that access it
+ * - Or stream should be synchronized before it is accessed from a different stream or on the host
+ *
  * TODO:
  *  - add constructor that takes pointer to hash_table to avoid allocations
  *  - extend interface to accept streams
```

Comment:
Is this TODO about streams still needed?

Reply:
`add constructor that takes pointer to hash_table to avoid allocations` still needs to be done